BackfillPowerBI.yml

```diff
@@ -0,0 +1,58 @@
+parameters:
+- name: DeleteAllOrganizationData
+  displayName: Delete all data in Power BI Simeon Data source
+  default: false
+  type: boolean
+
+- name: BackfillHours
+  displayName: Number of hours to include in backfill
+  default: 72
+  type: number
+
+name: $(Date:yyyyMMdd)$(Rev:.r)
+trigger: none
+pool:
+  vmImage: windows-latest
+resources:
+  repositories:
+  - repository: Reporting
+    type: github
+    name: simeoncloud/reporting
+    ref: refs/heads/feature/uploadToPowerBI
+    endpoint: simeoncloud
+jobs:
+- job: Job
+  timeoutInMinutes: 60
+  steps:
+  - checkout: Reporting
+
+  - pwsh: 
+      (Get-ChildItem -Filter 'Invoke-UploadPlanToPowerBI.ps1' -Recurse)[0].FullName | Import-Module -Force;
+      Connect-PowerBIAsServicePrincipal -AppId $env:AppId -AppSecret $env:Secret -Tenant $env:Tenant;
+      Clear-PowerBITable;
+    condition: and(succeeded(), ${{ parameters.DeleteAllOrganizationData }})
+    name: ClearPowerBITable
+    displayName: Clear all data from PowerBI
+    env:
+      AppId: $(AppId)
+      Secret: $(Secret)
+      Tenant: $(Tenant)
+
+  - pwsh: |
+      try {
+          (Get-ChildItem -Filter 'Invoke-UploadPlanToPowerBI.ps1' -Recurse)[0].FullName | Import-Module -Force
+          Connect-PowerBIAsServicePrincipal -AppId $env:AppId -AppSecret $env:Secret -Tenant $env:Tenant
+          Invoke-SimeonCloudReportingBackfill -OrganizationUri "$(System.CollectionUri)" -PersonalAccessToken $env:PersonalAccessToken -BackfillHours $env:BackfillHours
+      } catch {
+          Write-Error "There was an error"
```

Comment:
This will fail the job and the next two write-error lines won’t execute. I’d think you’d want this after the other two write-errors (in this case we actually do want it to fail the job)